src/main/host/descriptor/file.c

```diff
@@ -194,87 +183,118 @@ static char* _file_getPath(File* file, File* dir, const char* pathname) {
         abort();
     }
 
-    char* abspath = _file_getConcatStr(cwd, pathname);
+    char* abspath = _file_getConcatStr(cwd, '/', pathname);
     free(cwd);
     return abspath;
 }
 
-static char* _file_getTempPathTemplate(File* file, const char* pathname) {
-    char* abspath = NULL;
-    if (asprintf(&abspath, "%s/shadow-%i-tmpfd-%d-XXXXXX", pathname,
-                 (int)getpid(), _file_getFD(file)) < 0) {
-        error("asprintf could not allocate string for temp file, error %i: %s",
-              errno, strerror(errno));
-        abort();
-    }
-    return abspath;
-}
+#ifdef DEBUG
+#define CHECK_FLAG(flag)                                                                           \
+    if (flags & flag) {                                                                            \
+        if (!flag_str) {                                                                           \
+            asprintf(&flag_str, #flag);                                                            \
+        } else {                                                                                   \
+            flag_str = _file_getConcatStr(flag_str, '|', #flag);                                   \
```

Comment:
doesn't this leak the old `flag_str` buffer?

Reply:
Oops, good catch!